devicemodel/hw/pci/npk.c

```diff
@@ -244,11 +243,12 @@ static int pci_npk_init(struct vmctx *ctx, struct pci_vdev *dev, char *opts)
 	/* read the host NPK configuration space */
 	rc = snprintf(name, PATH_MAX, "%s/%s/config", NPK_DRV_SYSFS_PATH,
 			dent->d_name);
-	if (rc > PATH_MAX)
-		WPRINTF(("NPK device name too long\n"));
+	if (rc >= PATH_MAX || rc < 0)
+		WPRINTF(("NPK device name is invalid\n"));
+	closedir(dir);
```

Comment:
Haven't notice this change in mailing list.
Why move closedir to here?

Reply:
@ywan170   in current codebase, below statement is after closedir(dir), but it access 'dent->d_name',  so suggest  'closedir(dir)'  after related handler access.

 	rc = snprintf(name, PATH_MAX, "%s/%s/config", NPK_DRV_SYSFS_PATH,
			dent->d_name);